src/argus_htmx/themes/views.py

```diff
@@ -21,33 +21,43 @@ class ThemeListView(ListView):
 
     def setup(self, request, *args, **kwargs):
         super().setup(request, *args, **kwargs)
-        self.themes = sorted(get_theme_names())
+        self.themes = THEME_NAMES
 
     def get_queryset(self):
         return self.themes
 
     def post(self, request, *args, **kwargs):
-        theme = request.POST.get("theme")
-        if theme in self.themes:
-            request.session["theme"] = theme
+        prefs = request.user.get_preferences("argus_htmx")
+        form = prefs.FORMS["theme"](request.POST)
+        if form.is_valid():
+            theme = form.cleaned_data["theme"]
+            prefs.save_preference("theme", theme)
```

Comment:
This is a common pattern, can we extract/abstract it out?

Reply:
I'd love to.